src/Authentication/Authenticator.php

```diff
@@ -134,6 +134,17 @@ function getProfile()
     }
 
     /**
+     * will return a valid userProfile object for a pre-loaded profile
+     * (non-PHPdoc)
+     *
+     * @see \PhalconRest\Authentication\AuthenticatorInterface::getProfile()
+     */
+    function setProfile(\PhalconRest\Authentication\UserProfile $profile)
+    {
+        return $this->profile = $profile;
+    }
```

Comment:
BTW, would you also change this bit of documentation at `libraries/Authentication/UserProfile::loadProfile`? The method signature changed from the original method and the IDE is a bit dizzy on that.

I can't add it directly to this PR as I would need to fork your fork to send you a PR to be included in this PR.... lol

```php
    /**
     * load a full user profile based on three different filters
     *
     * ie.
     * "token = 'HACKYHACKERSON'"
     * "user_name = 'jjenkins'"
     * "token = 'A REALLY LONG STRING"
     *
     *
     * (non-PHPdoc)
     *
     * @see \PhalconRest\Authentication\UserProfile::loadProfileByToken()
     * @param string $mainFilter
     * @return bool
     * @throws HTTPException
     */
```